server/src/main/java/org/elasticsearch/action/admin/cluster/health/ClusterHealthResponse.java

```diff
@@ -299,7 +304,14 @@ public String toString() {
 
     @Override
     public RestStatus status() {
-        return isTimedOut() ? RestStatus.REQUEST_TIMEOUT : RestStatus.OK;
+        if (RestApiVersion.current() == RestApiVersion.V_7 && isTimedOut()) {
+            deprecationLogger.critical(DeprecationCategory.API, "cluster_health_request_timeout",
+                "HTTP status code for an internal cluster health timeout will change from 408 to 200 in a future major version");
+        }
+        if (Boolean.parseBoolean(System.getProperty("es.cluster_health.request_timeout_408", "false"))) {
+            return RestStatus.REQUEST_TIMEOUT;
```

Comment:
This returns `REQUEST_TIMEOUT` even if we didn't time out?

Reply:
Yes, indeed, the check wasn't put correctly. I need to add a test that sets `es.cluster_health.request_timeout_408` to catch such mistakes.

Reply:
++ I'd suggest adding another test to the `qa` project, that's generally the simplest way to control things like system properties.

Reply:
I've decided to make the property as not static, so we can set in a constructor in `ClusterHealthResponsesTests`. It seems to much faster and easier to test this way instead of creating another full blown integration test.

Reply:
The trouble with this is that if the system property is invalid, it'll just make cluster health explode (with a `500 Internal Server Error`) on a timeout. We should be stricter and not even let the node start up in that case.

I'm ok with passing the flag into the response constructor for a unit test but I think we still have to parse the property up front while the node is initialising, and I would very much prefer to have an end-to-end test that the opt-in actually works.